src/analysis/inner_product_space/l2_space.lean

```diff
@@ -419,6 +419,27 @@ end
   (b.to_orthonormal_basis : ι → E) = b :=
 orthonormal_basis.coe_mk _ _
 
+/-- For `e : hilbert_basis ι 𝕜 E` and `J : finset ι`, `e.partial_span J` is the span of
+the `e j`s for `j ∈ J`. -/
+def partial_span (b : hilbert_basis ι 𝕜 E) (J : finset ι) : submodule 𝕜 E :=
+span 𝕜 (J.image b)
```

Comment:
This definition feels a bit weird to me: it would work just as well as follows, right?
```suggestion
/-- For `e : hilbert_basis ι 𝕜 E` and `J : finset ι`, `e.partial_span J` is the span of
the `e j`s for `j ∈ J`. -/
def partial_span (b : ι -> E) (J : set ι) : submodule 𝕜 E :=
span 𝕜 (b '' J)
```
At which point, `partial_span b J` is a longer spelling than `span 𝕜 (b '' J)`